web3.js/test/connection.test.ts

```diff
@@ -903,6 +925,132 @@ describe('Connection', function () {
     ).to.be.rejectedWith(mockErrorMessage);
   });
 
+  describe('confirm transaction with mock timer', () => {
+    let clock: SinonFakeTimers;
+    beforeEach(() => {
+      clock = useFakeTimers();
+    });
+
+    afterEach(() => {
+      clock.restore();
+    });
+
+    it('confirm transaction - timeout expired', async () => {
+      const newSignature =
+        'w2Zeq8YkpyB463DttvfzARD7k9ZxGEwbsEw4boEK7jDp3pfoxZbTdLFSsEPhzXhpCcjGi2kHtHFobgX49MMhbWt';
+
+      await mockRpcMessage({
+        method: 'signatureSubscribe',
+        params: [newSignature, {commitment: 'finalized'}],
+        result: new Promise(() => {}),
+      });
+      const timeoutPromise = connection.confirmTransaction(newSignature);
+
+      clock.tick(60 * 1000);
+
+      await expect(timeoutPromise).to.be.rejectedWith(
+        `Signature ${newSignature} has expired: timeout exceeded.`,
+      );
+    });
+  });
+
+  it('confirm transaction - block height exceeded', async () => {
+    const blockHeightSignature =
+      '4oCEqwGrMdBeMxpzuWiukCYqSfV4DsSKXSiVVCh1iJ6pS772X7y219JZP3mgqBz5PhsvprpKyhzChjYc3VSBQXzG';
+
+    const txBlockHeightExpired = new TransactionExpiredBlockheightExceededError(
+      blockHeightSignature,
+    );
+
+    await mockRpcMessage({
+      method: 'signatureSubscribe',
+      params: [blockHeightSignature, {commitment: 'finalized'}],
+      result: new Promise(() => {}),
+    });
+
+    await mockRpcResponse({
+      method: 'getBlockHeight',
+      params: [{commitment: 'finalized'}],
+      value: 2,
+    });
+
+    await mockRpcResponse({
+      method: 'getBlockHeight',
+      params: [{commitment: 'finalized'}],
+      value: 3,
+    });
+
+    await mockRpcResponse({
+      method: 'getBlockHeight',
+      params: [],
+      value: 6,
+    });
+
+    await mockRpcMessage({
+      method: 'signatureSubscribe',
+      params: [blockHeightSignature, {commitment: 'finalized'}],
+      result: {err: txBlockHeightExpired},
+    });
```

Comment:
This test isn't supposed to test what happens when `signatureSubscribe` returns an error. It's only supposed to test what happens when `signatureSubscribe` _never_ gives a response, and then the `getBlockHeight` message returns `6`.

Reply:
Fixed this!